src/wallet/wallet.cpp

```diff
@@ -1344,6 +1344,21 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived,
         nFee = nDebit - nValueOut;
     }
 
+    // Create an output for the value taken from the transparent value pool by JoinSplits
+    if (nDebit > 0) {
+        CAmount nValueOut = 0;
+        for (const JSDescription & js : vjoinsplit) {
+            nValueOut += js.vpub_old;
```

Comment:
Let's say there is a net credit to the transparent value pool.  If there are no vouts, everything in vpub_new will flow to the miner.  So we do nothing.  If there is a vout (1) and if the address belongs to the wallet, than a credit entry is created... but if (2) the address does not belong to the wallet, and the transaction was signed/sent by the wallet (i.e. there was a vin), than a debit entry is created.